compiler/qsc/src/interpret/stateful.rs

```diff
@@ -361,86 +313,91 @@ impl Interpreter {
+            errors
+                .into_iter()
+                .map(|error| Error(WithSource::from_map(&package.sources, error).into()))
+                .collect::<Vec<_>>()
+        })?;
 
         let pass_errors = fragments
             .iter_mut()
-            .flat_map(|fragment| {
-                self.passes
-                    .run(self.store.core(), self.compiler.assigner_mut(), fragment)
-            })
+            .flat_map(|fragment| self.passes.run(core, &mut package.assigner, fragment))
             .collect::<Vec<_>>();
         if !pass_errors.is_empty() {
             return Err(pass_errors
                 .into_iter()
-                .map(|error| {
-                    LineError(WithSource::from_map(
-                        self.compiler.source_map(),
-                        error.into(),
-                    ))
-                })
+                .map(|error| Error(WithSource::from_map(&package.sources, error).into()))
                 .collect());
         }
 
-        // Items must be processed before top-level statements, so sort the fragments.
-        // Note that stable sorting is used here to preserve the order of top-level statements.
-        fragments.sort_by_key(|f| match f {
-            Fragment::Item(_) => 0,
-            Fragment::Stmt(_) => 1,
-        });
-
-        for fragment in fragments {
-            match fragment {
-                Fragment::Item(item) => match item.kind {
-                    qsc_hir::hir::ItemKind::Callable(callable) => {
-                        let callable = self.lower_callable_decl(&callable);
-
-                        self.callables
-                            .insert(qsc_eval::lower::lower_local_item_id(item.id), callable);
-                    }
-                    qsc_hir::hir::ItemKind::Namespace(..) => {}
-                    qsc_hir::hir::ItemKind::Ty(..) => {
-                        self.udts
-                            .insert(qsc_eval::lower::lower_local_item_id(item.id));
-                    }
-                },
-                Fragment::Stmt(stmt) => {
-                    let stmt_id = self.lower_stmt(&stmt);
-
-                    match self.eval_stmt(receiver, stmt_id) {
-                        Ok(value) => result = value,
-                        Err((error, call_stack)) => {
-                            let stack_trace = if call_stack.is_empty() {
-                                None
-                            } else {
-                                Some(self.render_call_stack(call_stack, &error))
-                            };
-
-                            return Err(vec![LineError(WithSource::from_map(
-                                self.compiler.source_map(),
-                                WithStack::new(error, stack_trace).into(),
-                            ))]);
-                        }
-                    }
-                }
+        // Partition list into items and statements since items are always
```

Comment:
Previously, we would do `sort_by_key` to process items before statements... I took this opportunity to just partition the list into two different ones since we're always going to process items and statements separately....